sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -160,8 +161,14 @@ def _build_table_properties_exp(
     ) -> t.Optional[exp.Properties]:
         properties: t.List[exp.Expression] = []
 
-        if storage_format:
-            properties.append(exp.FileFormatProperty(this=exp.Var(this=storage_format)))
+        if table_format and storage_format and self.dialect == "spark":
```

Comment:
What if I only have `table_format` but not `storage_format`? Technically if I specify `iceberg` in `table_format` I may want to use a whatever default storage format it comes with (i.e. parquet)

Reply:
This logic was for backwards compatibility for people who might have `storage_format=iceberg` in their models already. It takes `table_format` into account only if both are defined.

But I can improve this

Reply:
Yeah, I get that.  So far I see 2 scenario being covered:
1. Only `storage_format` is specified - this is for backward compatibilty
2. Both `table_format` and `storage_format` are specified - this is new

What about a case when there is `table_format` but no `storage_format`?

Reply:
You're right, that was missing. I've refactored the logic and added tests to cover the following scenarios for Spark:

- both `table_format` and `storage_format` defined
- just `table_format` defined
- just `storage_format` defined